wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -1010,21 +1010,37 @@ private void cleanup() {
 
         executor.shutdown();
 
+        boolean interrupted = false;
+        final ExecutorService destroyMachinesExecutor = Executors.newCachedThreadPool(
```

Comment:
I used `CachedThreadPool`, because it 
> creates new threads as needed, but will reuse previously constructed threads when they are available.

Do you have a suggestion?